src/side-navigation/__tests__/side-navigation.test.tsx

```diff
@@ -1219,6 +1221,88 @@ describe('SideNavigation', () => {
       expect(logo).toHaveAttribute('alt', 'logo');
     });
   });
+
+  describe('Items Header', () => {
+    it('renders select component as items header', () => {
+      const options = [
```

Comment:
I propose to keep it simple and render an empty select component. In the test it is enough to assert the select component can be found when it is passed as itemsHeader.

You can also add a second small test to ensure the findItemsHeader() returns null when it is not provided.